browser/ui/views/playlist/thumbnail_provider.cc

```diff
@@ -0,0 +1,137 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/playlist/thumbnail_provider.h"
+
+#include "base/containers/flat_map.h"
+#include "base/containers/lru_cache.h"
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/no_destructor.h"
+#include "base/task/thread_pool.h"
+#include "base/task/thread_pool/thread_pool_instance.h"
+#include "brave/browser/playlist/playlist_service_factory.h"
+#include "brave/browser/playlist/playlist_tab_helper.h"
+#include "brave/components/playlist/browser/playlist_constants.h"
+#include "brave/components/playlist/browser/playlist_service.h"
+#include "net/base/filename_util.h"
+
+namespace {
+
+using ItemImageCache = base::LRUCache<std::string /*item_id*/, gfx::Image>;
+
+ItemImageCache& GetInMemoryCache(playlist::PlaylistService* service) {
```

Comment:
Using in-memory cache could increase run-time memory usage.
If thumbnail is already cached(item has valid `thumbnail_path`), I think using that cache would be sufficient.
and it'll be fetched very quickly.
Doubt that using this in-memory cache could give much benefit over its memory usage.

Reply:
As you noticed, this in-memory cache saves not only the cost of file read, but also network

Reply:
NVM, I think it could be useful before item has its thumbnail in local disk.